jekyll/_cci2/rotate-project-ssh-keys.adoc

```diff
@@ -0,0 +1,110 @@
+
+When using project SSH keys, CircleCI holds the private key, and the target system (typically a VCS, machine or other SSH based system) holds the matching public key via its authorization system. To rotate SSH keys both the private and public keys need to be removed.
+
+[#github-projects]
+== Github projects
+
+Go to **Project Settings > SSH Keys** to view SSH keys set up for your project.
+
+[#rotate-a-deploy-key-github]
+=== Rotate a deploy key
+
+. Take note of the current key information to rotate, including the fingerprint. You can also click the keyname to open the related GitHub page that should list the public key.
+. Delete the deploy key by clicking the **X**.
+. Add a new deploy key by clicking **Add Deploy Key**.
+. Go to GitHub’s repository project settings to delete the matching public key. The github URL is typically `https://github.com/<org>/<project>/settings/keys`, or you may already have the page open if you clicked on the keyname in step 1. The keys are named `CircleCI`. Removing any key titled `CircleCI` created before the rotation is recommended. The new public SSH key will be automatically added once the old key is deleted.
+
+[#rotate-a-user-key-github]
+=== Rotate a user key
+
+If you have set up user keys for your project, follow these steps:
+
+. Take note of the current key information to rotate, including the fingerprint. You can also click the keyname to open the related Github page that should list the public key.
+. Delete the deploy key by clicking the **X**.
+. Add a new user key by clicking **Add User Key**.
+. Go to GitHub’s User settings to delete the matching public key. The github URL is typically `https://github.com/settings/keys`or you may already have the page open if you clicked on the keyname in step 1. Removing any key titled `CircleCI` created before this rotation is recommended. The new public SSH key will be automatically added once the old key is deleted (page may need to be refreshed to view the key).
+
+CAUTION: The user key name contains the project name, however, a user key may give access to more projects within the organization.
+
+NOTE: If using organization SSO, you can take note of which one is currently authorized. For the newly created key, if using organization SSO, and access is needed, you will need to reauthorize it.
+
+[#rotate-an-additional-SSH-key-github]
+=== Rotate an additional SSH key
+
+If you are using additional SSH keys in your project, follow these steps:
+
+. Take note of the existing key to know which target system it is used for. This could be a VCS, a machine or other SSH based system.
+. Delete the deploy key by clicking the **X**.
+. Remove the authorization to use that key (the public key) from the target system. Typically, the public key will be authorized in the target system, and the configuration varies depending on the target system.
+. Recreate the key and the authorization in the target system.
```

Comment:
Are we supposed to include an additional step here for "Add SSH Key" within the CircleCI project?

Reply:
yes! I realized that too. The 3 statements are missing the step to do it in cci UI. It is the same as explained for gitlab additional keys
**Add the private key by clicking Add SSH Key. Paste in the private key you created in the step above and click Add SSH Key again to finalize.**

Reply:
@sbeaulie Can they do this in the CCI web app straight after deleting the original additional SSH private key, before leaving the app to go an delete the public key from the target? 

And are the steps as follows:

. create a new SSH key using `ssh-keygen -t ed25519 -C your_email@example.com` or however they want to generate it
. click ADD SSH key
. As Hostname you can use the same value you had for the old key in step 1
. click Add SSH Key again to finalize